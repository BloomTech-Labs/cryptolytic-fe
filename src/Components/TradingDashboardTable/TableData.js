import React from "react";
import { withStyles, makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  binanceColor: {
    color: "rgb(78, 185, 255)",

  },
  coinbaseColor: {
    color: "rgb(83, 207, 215)",
 
  },
  bitfinex: {
    color: "rgb(98, 227, 171)",
  },
  positive: {
    color: "green",
  },
  negative: {
    color: "red"
  },
  tradingPair: {
    display: "flex",
    marginLeft: "33%",

  }
});

const TableData = ({ item, positive, negative, icon1, icon2 }) => {
  const classes = useStyles();

  if (positive && item >= 0) {
    return <td className={positive}>{item + " %"}</td>;
  } else if (negative && negative) {
    return <td className={negative}>{item + " %"}</td>;
  } else if (item === "Binance") {
    return <td className={classes.binanceColor}>{item}</td>;
  } else if (item === "Coinbase Pro") {
    return <td className={classes.coinbaseColor}>{item}</td>;
  } else if (item === "Bitfinex") {
    return <td className={classes.bitfinex}>{item}</td>;
  } else if (icon1 && icon2) {
    return (
      <td className={classes.tradingPair}>
        <img src={icon1} />
        <img src={icon2} />

        {item}
      </td>
    );
  } else {
    return <td>{item}</td>;
  }
};

export default TableData;
