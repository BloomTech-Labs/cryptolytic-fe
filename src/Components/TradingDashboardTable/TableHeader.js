import React from "react";
import zIndex from "@material-ui/core/styles/zIndex";

const TableHeader = ({ header }) => {
  if (header.includes("_")) {
    header = header
      .replace(/\_/, " ")
      .split(" ")
      .map(word => {
        return word.charAt(0).toUpperCase() + word.substr(1);
      })
      .join(" ");
  }

  return (
    <th
      style={{
        padding: "0.5em 0",
        position: "sticky",
        top: 0,
        background: "#191E23"
      }}
    >
      {header}
    </th>
  );
};

export default TableHeader;
