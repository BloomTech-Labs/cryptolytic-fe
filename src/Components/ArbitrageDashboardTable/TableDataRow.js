import React from "react";
import TableData from "./TableData";
import { getIcons } from "../../Utilities/tradingPairIcons";

import { makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";

const useStyles = makeStyles({
  tablerow: {
    background: "rgba(0, 0, 0, 0.8)",
    color: "rgb(255, 255, 255)"
  },
  positive: {
    color: "green",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  },
  negative: {
    color: "red",
    padding: "0.5em 0",
    borderBottom: "1px solid rgba(35, 32, 44, 0.9)"
  }
});

const TableDataRow = ({ dataRow }) => {
  console.log(dataRow, `arb`);
  const classes = useStyles();

  dataRow.buy_exchange = dataRow.buy_exchange
    .replace(/\_/, " ")
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  dataRow.sell_exchange = dataRow.sell_exchange
    .replace(/\_/, " ")
    .split(" ")
    .map(word => {
      return word.charAt(0).toUpperCase() + word.substr(1);
    })
    .join(" ");

  dataRow.trading_pair = dataRow.trading_pair.toUpperCase();
  dataRow.trading_pair = dataRow.trading_pair.replace(/\_/, "/");

  if (String(dataRow.price_difference).length <= 3) {
    dataRow.price_difference =
      "$ " + String(dataRow.price_difference.toFixed(2));
  } else {
    dataRow.price_difference = "$ " + String(dataRow.price_difference);
  }

  if (String(dataRow.arbritage_percentage).length <= 3) {
    dataRow.arbritage_percentage = dataRow.arbritage_percentage.toFixed(2);
  }

  const data = Object.values(dataRow);

  const icons = getIcons(dataRow.trading_pair);
  return (
    <TableRow className={classes.tablerow}>
      {data.map((item, i) => {
        if (i === 6) {
          return (
            <TableData
              item={item}
              positive={classes.positive}
              negative={classes.negative}
            />
          );
        } else if (i === 4) {
          return (
            <TableData item={item} icon1={icons.icon1} icon2={icons.icon2} />
          );
        }
        return <TableData item={item} />;
      })}
    </TableRow>
  );
};

export default TableDataRow;
