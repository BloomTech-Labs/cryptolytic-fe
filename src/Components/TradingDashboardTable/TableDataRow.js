import React from "react";
import TableData from "./TableData";
import { getIcons } from "../../Utilities/tradingPairIcons";

import { withStyles, makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";

const useStyles = makeStyles({
  tablerow: {
    background: "rgba(0, 0, 0, 0.8)",
    color: "rgb(255, 255, 255)",
    borderBottom: "0.5px solid rgba(35, 32, 44, 0.8)"
  },
  positive: {
    color: "green"
  },
  negative: {
    color: "red"
  }
});

const TableDataRow = ({ dataRow }) => {
  const classes = useStyles();

  dataRow.exchange =
    dataRow.exchange.charAt(0).toUpperCase() + dataRow.exchange.substr(1);
  dataRow.trading_pair = dataRow.trading_pair.toUpperCase();
  dataRow.trading_pair = dataRow.trading_pair.replace(/\_/, "/");

  const data = Object.values(dataRow);

  const icons = getIcons(dataRow.trading_pair);

  return (
    <TableRow className={classes.tablerow}>
      {data.map((item, i) => {
        if (i === 6) {
          return (
            <TableData
              item={item}
              positive={classes.positive}
              negative={classes.negative}
            />
          );
        } else if (i === 4) {
          return (
            <TableData item={item} icon1={icons.icon1} icon2={icons.icon2} />
          );
        }
        return <TableData item={item} />;
      })}
    </TableRow>
  );
};

export default TableDataRow;
