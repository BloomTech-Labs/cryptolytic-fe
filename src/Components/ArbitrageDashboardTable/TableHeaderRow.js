import React from "react";
import TableHeader from "./TableHeader";

import { makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";

const useStyles = makeStyles({
  tablerow: {
    background: "#23282D",
    color: "rgb(112, 112, 112)"
  }
});

const TableHeaderRow = ({ headerData }) => {
  headerData = headerData.map(header => {
    return header.replace(/_/, " ");
  });

  const classes = useStyles();

  return (
    <TableRow className={classes.tablerow}>
      {headerData.map(header => (
        <TableHeader header={header} />
      ))}
    </TableRow>
  );
};

export default TableHeaderRow;
