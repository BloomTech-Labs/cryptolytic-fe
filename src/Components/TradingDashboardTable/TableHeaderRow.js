import React from "react";
import TableHeader from "./TableHeader";

import { withStyles, makeStyles } from "@material-ui/core/styles";
import TableRow from "@material-ui/core/TableRow";
import zIndex from "@material-ui/core/styles/zIndex";

const useStyles = makeStyles({
  tablerow: {
    background: "#23282D",
    color: "rgb(112, 112, 112)"
  }
});

const TableHeaderRow = ({ headerData }) => {
  const classes = useStyles();

  headerData = headerData.map(header => {
    return header.charAt(0).toUpperCase() + header.substr(1);
  });

  return (
    <TableRow className={classes.tablerow}>
      {headerData.map(header => (
        <TableHeader className={classes.tableHead} header={header} />
      ))}
    </TableRow>
  );
};

export default TableHeaderRow;
