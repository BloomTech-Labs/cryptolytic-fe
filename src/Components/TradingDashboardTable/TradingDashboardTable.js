import React, { useState, useEffect } from "react";
import axios from "axios";
import { css } from "@emotion/core";
import PropagateLoader from "react-spinners/PropagateLoader";
import TableHeaderRow from "./TableHeaderRow";
import TableDataRow from "./TableDataRow";
import { makeStyles } from "@material-ui/core/styles";
import TableContainer from "@material-ui/core/TableContainer";
import Table from "@material-ui/core/Table";
import TableHead from "@material-ui/core/TableHead";
import TableBody from "@material-ui/core/TableBody";

const useStyles = makeStyles({
  table: {
    marginLeft: "20vw",
    width: "70%",
    background: "rgba(35, 32, 44, 0.8)",
    boxShadow: "0px 4px 4px rgba(0, 0, 0, 0.25)",
    "&::-webkit-scrollbar-thumb": {
      backgroundColor: "rgba(255,255,255,1.0)",
      outline: "1px solid slategrey"
    }
  },
  tableContainer: {
    maxHeight: 800,

    "&::-webkit-scrollbar": {
      display: "none"
    }
  }
});

const override = css`
  display: flex;
  justify-content: center;
  position: relative;
  margin: 30vh 0 50vh 10vw;
`;

const TradingDashboardTable = () => {
  const classes = useStyles();

  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    axios
      .get("https://cryptolytic-starter.herokuapp.com/trading")
      .then(res => {
        setData(res.data);
        setTimeout(() => {
          setLoading(false);
        }, 2000);
      })
      .catch(error => {
        console.log("error", error);
      });
  }, []);

  const tradingData = data.map(obj => {
    let {
      exchange,
      trade_time,
      trade_price,
      period,
      percentage,
      trading_pair
    } = obj;

    let date, time;
    trade_time = new Date(obj.trade_time * 1000);

    date = trade_time.toLocaleDateString();
    time = trade_time.toLocaleTimeString();

    return {
      date,
      time,
      period,
      exchange,
      trading_pair,
      trade_price,
      percentage
    };
  });

  const headerData = [];

  for (let key in tradingData[0]) {
    headerData.push(key);
  }

  if (loading) {
    return (
      <PropagateLoader
        css={override}
        size={25}
        color={
          "linear-gradient(93.16deg, #4EB9FF 19.25%, #53CFD7 45.13%, #5DDCB7 67.95%, #62E3AB 82.93%);"
        }
        loading={loading}
      />
    );
  } else {
    return (
      <TableContainer className={classes.tableContainer}>
        <Table stickyHeader aria-label='sticky table' className={classes.table}>
          <TableHead>
            <TableHeaderRow headerData={headerData} />
          </TableHead>
          <TableBody>
            {tradingData.map(items => (
              <TableDataRow dataRow={items} />
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    );
  }
};

export default TradingDashboardTable;
