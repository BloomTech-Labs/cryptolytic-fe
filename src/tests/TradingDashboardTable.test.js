import React from "react";
import { MemoryRouter } from "react-router";
///
import TradingDashboardTable from "../Components/TradingDashboardTable/TradingDashboardTable";
import TableHeaderRow from "../Components/TradingDashboardTable/TableHeaderRow";
import TableDataRow from "../Components/TradingDashboardTable/TableDataRow";
///
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";

test("Trading DashboardTable components initialized without crashing", () => {
    render(<TableHeaderRow headerData={["Header", "Test"]} />);
    render(
      <TableDataRow
        dataRow={{
          buy_exchange: "binance",
          sell_exchange: "bitfinex",
          trading_pair: "usd_btc",
          exchange: "binance"
        }}
      />
    );
});

describe("<TradingDashboardTable /> matches verified SnapShot", () => {
  it("should match snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
            <TradingDashboardTable />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
