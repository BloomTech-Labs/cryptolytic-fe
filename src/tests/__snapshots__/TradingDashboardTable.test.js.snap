// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TradingDashboardTable /> matches verified SnapShot should match snapshot 1`] = `
<div
  className="MuiTableContainer-root makeStyles-tableContainer-17"
>
  <table
    className="MuiTable-root makeStyles-table-16"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root makeStyles-tablerow-22 MuiTableRow-head"
      />
    </thead>
    <tbody
      className="MuiTableBody-root"
    />
  </table>
</div>
`;
